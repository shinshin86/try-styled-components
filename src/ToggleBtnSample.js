import React, { Component } from 'react';
import styled from 'styled-components';

const Container = styled.div`
  margin: 30px;
  width: 100%;
  border: solid 1px #ccc;
`;

const ToggleBtn = styled.button`
  background: #134a63;
  color: #eee;
  width: 300px;
  height: 50px;
  margin: 30px;
  ${({ show }) =>
    show &&
    `
    background: #DE5065;
  `};
`;

export default class ToggleBtnSample extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isShow: false
    };
    this.handleToggleBtn = this.handleToggleBtn.bind(this);
  }

  handleToggleBtn() {
    this.setState({ isShow: !this.state.isShow });
  }

  render() {
    return (
      <Container>
        <ToggleBtn show={this.state.isShow} onClick={this.handleToggleBtn}>
          Toggle Button
        </ToggleBtn>
      </Container>
    );
  }
}
