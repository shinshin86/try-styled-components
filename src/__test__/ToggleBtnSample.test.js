import React from 'react';
import renderer from 'react-test-renderer';
import ToggleBtnSample from '../ToggleBtnSample';
import 'jest-styled-components';

test('ToggleBtnSample Snapshot', () => {
  const tree = renderer.create(<ToggleBtnSample />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ToggleBtnSample have border', () => {
  const tree = renderer.create(<ToggleBtnSample />).toJSON();
  expect(tree).toHaveStyleRule('border', 'solid 1px #ccc');
});
