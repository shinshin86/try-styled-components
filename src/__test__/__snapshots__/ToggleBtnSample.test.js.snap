// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleBtnSample Snapshot 1`] = `
.c0 {
  margin: 30px;
  padding: 30px;
  border: solid 1px #ccc;
}

.c2 {
  margin: 100px;
  border: solid 1px #ccc;
}

.c1 {
  background: #134a63;
  color: #eee;
  width: 300px;
  height: 50px;
  margin: 30px;
}

.c3 {
  color: #de5;
  width: 250px;
  height: 70px;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    Toggle Button
  </button>
  <div
    className="c2"
  >
    <button
      className="c3 c1"
      onClick={[Function]}
    >
      Toggle Button
    </button>
  </div>
</div>
`;
